src/meta/src/hummock/compaction_schedule_policy.rs

```diff
@@ -0,0 +1,722 @@
+        self.compactors.push(Arc::new(Compactor::new(
+            context_id,
+            tx,
+            max_concurrent_task_number,
+        )));
+        tracing::info!("Added compactor session {}", context_id);
+        rx
+    }
+
+    fn random_compactor(&mut self, _compact_task: Option<&CompactTask>) -> Option<Arc<Compactor>> {
+        if self.compactors.is_empty() {
+            return None;
+        }
+
+        let compactor_index = rand::thread_rng().gen::<usize>() % self.compactors.len();
+        let compactor = self.compactors[compactor_index].clone();
+        Some(compactor)
+    }
+
+    fn remove_compactor(&mut self, context_id: HummockContextId) {
+        self.compactors.retain(|c| c.context_id() != context_id);
+        tracing::info!("Removed compactor session {}", context_id);
+    }
+
+    fn report_compact_task(&mut self, _context_id: HummockContextId, _task: &CompactTask) {}
+}
+
+#[derive(Default)]
+pub struct LeastPendingBytesPolicy {
+    // These two data structures must be consistent.
+    // We use `(pending_bytes, context_id)` as the key to dedup compactor with the same pending
+    // bytes.
+    pending_bytes_to_compactor: BTreeMap<(usize, HummockContextId), Arc<Compactor>>,
+    compactor_to_pending_bytes: HashMap<HummockContextId, usize>,
+}
+
+impl LeastPendingBytesPolicy {
+    pub fn new() -> Self {
+        Self {
+            ..Default::default()
+        }
+    }
+
+    fn update_compactor(
+        &mut self,
+        context_id: HummockContextId,
+        old_bytes: usize,
+        compact_task: Option<&CompactTask>,
+        assign_task: bool,
+    ) -> Arc<Compactor> {
+        let task_pending_bytes = match compact_task {
+            Some(task) => task.input_file_size as usize,
+            None => 0,
+        };
+        let new_bytes = if assign_task {
+            old_bytes + task_pending_bytes
+        } else {
+            debug_assert!(old_bytes >= task_pending_bytes);
+            old_bytes - task_pending_bytes
+        };
+        // The element must exist.
+        let compactor = self
+            .pending_bytes_to_compactor
+            .remove(&(old_bytes, context_id))
+            .unwrap();
+        self.pending_bytes_to_compactor
+            .insert((new_bytes, context_id), compactor.clone());
+        *self
+            .compactor_to_pending_bytes
+            .get_mut(&context_id)
+            .unwrap() = new_bytes;
+        compactor
+    }
+}
+
+#[async_trait::async_trait]
+impl CompactionSchedulePolicy for LeastPendingBytesPolicy {
+    async fn next_idle_compactor<S: MetaStore>(
+        &mut self,
+        hummock_manager: &HummockManager<S>,
+        compact_task: Option<&CompactTask>,
+    ) -> Option<Arc<Compactor>> {
+        let mut next_idle_compactor = None;
+        for ((pending_bytes, context_id), compactor) in &self.pending_bytes_to_compactor {
+            if hummock_manager.get_assigned_tasks_number(*context_id).await
```

Comment:
This operation would acquire a read-write lock, which may execute an IO operation on ETCD.
I think we shall maintain another copy in this structure. In another word, we store mapping context_id -->  assigned_tasks_number  in `LeastPendingBytesPolicy` or some Context.

Reply:
We can also store it in `CompactorManager` because this PR would `report_compact_task` to `CompactorManager` , in which we can update assigned_tasks_number.